### PR TITLE
Add ISO 8583 tcp inbound

### PR DESCRIPTION
**Pull request by RivinduM**

### CODE REVIEW

File: iso8583-tcp-integration-and-message-transformation/backend_service/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[package]
+org = "wso2"
+name = "backend_service"
+version = "0.1.0"
+distribution = "2201.10.1"
```
Comment (VimukthiRajapaksha) [inline iso8583-tcp-integration-and-message-transformation/backend_service/Ballerina.toml:5]: Isn't it better to add a few keywords?

File: iso8583-tcp-integration-and-message-transformation/backend_service/records.bal [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+    string PointOfServiceCaptureCode?;
+    string CardSequenceNumber?;
+    string ForwardingInstitutionIdentificationCode?;
+|};
+
+public type MTI_0100 record {|
+    string MTI = "0100";
+    @constraint:String {
+         pattern: {
+            value: re `^\d+`,
```
Comment (VimukthiRajapaksha) [inline iso8583-tcp-integration-and-message-transformation/backend_service/records.bal:174]: Is it possible to get these values from constants?

File: iso8583-tcp-integration-and-message-transformation/backend_service/service.bal [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                ActiveCurrencyAndAmount_SimpleType.ActiveCurrencyAndAmount_SimpleType;
+            if amount > 0d {
+                return http:ACCEPTED;
+            } else if isopacs008.FIToFICstmrCdtTrf.CdtTrfTxInf[0].PmtTpInf?.LclInstrm?.Prtry.toString().startsWith("31") {
+                return http:ACCEPTED;
+            } else {
+                return http:BAD_REQUEST;
+            }
+        } else {
+            return http:BAD_REQUEST;
```
Comment (VimukthiRajapaksha) [inline iso8583-tcp-integration-and-message-transformation/backend_service/service.bal:42]: Don't we need to log this error?

File: iso8583-tcp-integration-and-message-transformation/iso8583_tcp_inbound/main.bal [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    localHost: inboundConfig.host
+};
+
+# TCP service to receive inbound messages of ISO 8583.
+service on new tcp:Listener(inboundConfig.port, listenerConfig) {
+
+    // This remote method is invoked when the new client connects to the server.
+    remote function onConnect(tcp:Caller caller) returns tcp:ConnectionService {
+        io:println("Client connected to ISO 8583 TCP server ", inboundConfig.host, ":", inboundConfig.port, 
+            "caller remote port: ", caller.remotePort);
```
Comment (VimukthiRajapaksha) [inline iso8583-tcp-integration-and-message-transformation/iso8583_tcp_inbound/main.bal]:
```suggestion
            " caller remote port: ", caller.remotePort);
```

File: iso8583-tcp-integration-and-message-transformation/iso8583_tcp_inbound/transaction_handler.bal [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+# Handles inbound transactions and return response.
+#
+# + data - byte array of the incoming message
+# + return - byte array of the response message
+public function handleInbound(byte[] & readonly data) returns byte[]|error {
+
+    string hex = array:toBase16(data);
+    log:printDebug("[Transaction Handler] Received message from the network driver: " + hex);
+    log:printDebug("[Transaction Handler] Recieived data string: " + data.toString());
```
Comment (VimukthiRajapaksha) [inline iso8583-tcp-integration-and-message-transformation/iso8583_tcp_inbound/transaction_handler.bal]: Shall we move this to L34?

**Review by VimukthiRajapaksha — DISMISSED** [review]

File: iso8583-tcp-integration-and-message-transformation/backend_service/Ballerina.toml (hunk 1 above)
Comment (RivinduM) [reply iso8583-tcp-integration-and-message-transformation/backend_service/Ballerina.toml:5]: added with [add2697](https://github.com/wso2/solutions-financial-demos/pull/1/commits/add2697ad0cbcfc92467395b9f0d49e58ce18ebe)

File: iso8583-tcp-integration-and-message-transformation/backend_service/records.bal (hunk 2 above)
Comment (RivinduM) [reply iso8583-tcp-integration-and-message-transformation/backend_service/records.bal:174]: seems it's not possible to define this as a constant

File: iso8583-tcp-integration-and-message-transformation/backend_service/service.bal (hunk 3 above)
Comment (RivinduM) [reply iso8583-tcp-integration-and-message-transformation/backend_service/service.bal:42]: fixed with [add2697](https://github.com/wso2/solutions-financial-demos/pull/1/commits/add2697ad0cbcfc92467395b9f0d49e58ce18ebe)

File: iso8583-tcp-integration-and-message-transformation/iso8583_tcp_inbound/main.bal (hunk 4 above)
Comment (RivinduM) [reply iso8583-tcp-integration-and-message-transformation/iso8583_tcp_inbound/main.bal]: fixed with [add2697](https://github.com/wso2/solutions-financial-demos/pull/1/commits/add2697ad0cbcfc92467395b9f0d49e58ce18ebe)

File: iso8583-tcp-integration-and-message-transformation/iso8583_tcp_inbound/transaction_handler.bal (hunk 5 above)
Comment (RivinduM) [reply iso8583-tcp-integration-and-message-transformation/iso8583_tcp_inbound/transaction_handler.bal]: fixed with [add2697](https://github.com/wso2/solutions-financial-demos/pull/1/commits/add2697ad0cbcfc92467395b9f0d49e58ce18ebe)

**Review by VimukthiRajapaksha — APPROVED** [review]

**Review by Ashi1993 — APPROVED** [review]

**Pull request merged**